src/doc/playback_tests.cpp

```diff
@@ -464,39 +505,361 @@ TEST(Playback, PingPongWithInnerReverse)
   EXPECT_FALSE(play.isStopped());
 }
 
+// OnePingPongInsideOther series
+std::vector<int> goRight(std::vector<int> range) {
+  std::vector<int> out;
+  if (range[0] > range[1])
+    return out;
+  for (int i=range[0]; i<range[1]+1 ; ++i)
+    out.push_back(i);
+  return out;
+}
+
+std::vector<int> goLeft(std::vector<int> range) {
+  std::vector<int> out;
+  if (range[0] > range[1])
+    return out;
+  for (int i=range[1]; i>range[0]-1 ; --i)
+    out.push_back(i);
+  return out;
+}
+
+void concat(std::vector<int>& a, std::vector<int>& b)
+{
+  if (b.empty())
+    return;
+  for (int i=0; i<b.size(); ++i)
+    a.push_back(b[i]);
+}
```

Comment:
Added `&`, `const`, and no need to check for empty (as size will be 0 anyway):
```suggestion
static void concat(std::vector<int>& a, const std::vector<int>& b)
{
  for (size_t i=0; i<b.size(); ++i)
    a.push_back(b[i]);
}
```